### ISSUE TITLE
Missing HasForeignKey when referencing a navigation property with a unique constraint. (EF Core 7 preview)

### ISSUE THREAD

**Issue by TimKras** [ErikEJ/EFCorePowerTools#1554]
Describe what is not working as expected.

### Provide steps to reproduce

The tool is generating this code:
```
          entity.HasOne(d => d.VrijstellingsCodeBHNavigation).WithMany(p => p.Klant)
                .HasPrincipalKey(p => p.VrijstellingsCodeBH)
                .HasConstraintName("FK_Klant_BTWVrijstellingsCode");
```
At runtime I get this exception:
![image](https://user-images.githubusercontent.com/5257192/197607096-e61935f6-ee9d-4d57-8846-c94cd2dda720.png)
Indeed, the column does not exists in the database.

When adding manual this line, everything is working like it should:
```
            entity.HasOne(d => d.VrijstellingsCodeBHNavigation).WithMany(p => p.Klant)
                .HasForeignKey(p => p.VrijstellingsCodeBH)
                .HasPrincipalKey(p => p.VrijstellingsCodeBH)
                .HasConstraintName("FK_Klant_BTWVrijstellingsCode");
```

--> it looks like the HasForeignKey() is missing.

### Provide technical details

- EF Core version in use: EF Core 7

- Is Handlebars used: no

- Is T4 used: no

- Is .dacpac used: no

- EF Core Power Tools version: latest version

- Database engine: SQL Server

- Visual Studio version: Visual Studio 2022

**Comment by ErikEJ** [comment]
What happens when you do not use t4?

I need a full SQL schema to investigate

**Comment by TimKras** [comment]
@ErikEJ 
I have created an example project to demonstrate the problem. See attached, there is also a script to generate the database.

It is not related to T4 template usage.
The problem does only occur when using '`EF Core 7 (preview)`' in combination with '`Use DataAnnotation attributes to configure the model`'.
When unchecking the '`Use DataAnnotation attributes`' option, the `HasForeignKey` statement is generated (and the application is working).
[EFTest.zip](https://github.com/ErikEJ/EFCorePowerTools/files/9857912/EFTest.zip)

**Comment by ErikEJ** [comment]
Yes, I am able to repro - and EF Core 6 reverse engineering generates this with DataAnnotations enabled:

```csharp
 entity.HasOne(d => d.ColorCodeNavigation)
                    .WithMany(p => p.Car)
                    .HasPrincipalKey(p => p.ColorCode)
                    .HasForeignKey(d => d.ColorCode)
                    .HasConstraintName("FK_Car_Color");

```

**Comment by ErikEJ** [comment]
I have filed a bug in the EF Core repo: https://github.com/dotnet/efcore/issues/29418

**Comment by TimKras** [comment]
Confirmed fixed in 7.0.3

**Comment by ErikEJ** [comment]
@TimKras Thanks for the update, EF Core Power Tools uses 7.0.3 now
